bridges/scala-native-0.4/src/main/scala/bloop/scalanative/NativeBridge.scala

```diff
@@ -37,6 +37,7 @@ object NativeBridge {
     }
     val nativeLTO = config.mode match {
       case LinkerMode.Debug => build.LTO.none
+      case LinkerMode.Release if bloop.util.CrossPlatform.isMac => build.LTO.full
       case LinkerMode.Release => build.LTO.thin
```

Comment:
Seems pretty weird to do something different for Mac than others? Is that something the sbt plugin does as well?

Reply:
It's mostly related to LLVM compilation errors happening only on MacOS with LTO=thin  
https://github.com/scala-native/scala-native/pull/3833 No idea how to fix it currently. 

I have no idea about rest of the codebase and I guess there is no way to configure LTO directly, however we might detect the LTO using env variables? There is a dedicated method to do it, it's used in the sbt plugin. 
https://github.com/scala-native/scala-native/blob/4b9f6314a90336d448e0b5cb0fdcb3f122f5c47b/tools/src/main/scala/scala/scalanative/build/Discover.scala#L25

It might be beneficial, because LTO.full can be sometimes slow, so maybe defaulting to LTO.none would be a good choice? 
Also, in some configurations LTO=thin might work on MacOS, (see our nightly builds https://github.com/scala-native/scala-native/actions/runs/8226709422/job/22494744082 where we use LLVM linker `lld` and LLVM distribution of `clang` instead of the one shipped with MacOS (it does not work on M1 chips though)

Reply:
Hmm... so for now only change in tests? OR default to none, which would be better?

Reply:
Defaulting to none is the safest option in my opinion.

Reply:
Switched to none as a default